pkgs/misc/vim-plugins/overrides.nix

```diff
@@ -643,6 +644,23 @@ self: super: {
     dependencies = with self; [ telescope-nvim ];
   });
 
+  tup =
+    let
+      # Based on the comment at the top of the file
+      ftdetect = builtins.toFile "tup.vim" ''
+        au BufNewFile,BufRead Tupfile, *.tup set filetype=tup
+      '';
+    in
+    buildVimPluginFrom2Nix {
+      inherit (tup) pname version src;
+      preInstall = ''
+        mkdir vim-plugin && cd vim-plugin
+        mkdir syntax && cp ../contrib/syntax/tup.vim ./syntax/tup.vim
+        mkdir ftdetect && cp "${ftdetect}" ./ftdetect/tup.vim
```

Comment:
```suggestion
        mkdir -p vim-plugin/syntax vim-plugin/ftdetect 
        cp ../contrib/syntax/tup.vim vim-plugin/syntax/tup.vim
        cp "${ftdetect}" vim-plugin/ftdetect/tup.vim
```

Reply:
I've had to modify these changes, as using cd is needed to avoid copying all of Tup into the vim plugin (among a few minor fixes).